inst/stan/epinowcast.stan

```diff
@@ -22,78 +22,128 @@ data {
   array[s] int csl; // cumulative version of the above
   array[s] int sg; // how snapshots are related
   int dmax; // maximum possible report date
+
   // Observations
   array[s, dmax] int obs; // obs for each primary date (row) and report date (column)
   array[n] int flat_obs; // obs stored as a flat vector
   array[t, g] int latest_obs; // latest obs for each snapshot group
+
+  // Expectation model
+  array[2] real eobs_lsd_p; // Standard deviation for expected final observations
+
   // Reference day model
-  int refp_fnrow; // how many unique pmfs there are
-  array[s] int refp_findex; // how each date links to a pmf
-  int refp_fncol; // number of effects to apply
-  matrix[refp_fnrow, refp_fncol + 1] refp_fdesign; // design matrix for pmfs
-  int refp_rncol; // number of standard deviations to use for pooling
-  matrix[refp_fncol, refp_rncol + 1] refp_rdesign; // Pooling pmf design matrix 
-  int model_refp; // parametric distribution (0 = none, 1 = exp, 2 = lognormal, 3 = gamma, 4 = loglogistic)
+  int model_refp;
+  int refp_fnrow;
+  array[s] int refp_findex;
+  int refp_fncol;
+  matrix[refp_fnrow, refp_fncol + 1] refp_fdesign;
+  int refp_rncol;
+  matrix[refp_fncol, refp_rncol + 1] refp_rdesign; 
+  array[2] real refp_mean_int_p;
+  array[2] real refp_sd_int_p;
+  array[2] real refp_mean_beta_sd_p;
+  array[2] real refp_sd_beta_sd_p; 
+
   // Reporting day model
-  int model_rep; // Reporting day model in use
+  int model_rep;
   int rep_t; // how many reporting days are there (t + dmax - 1)
-  int rep_fnrow; // how many unique reporting days are there
-  array[g, rep_t] int rep_findex; // how each report date links to a sparse report effect
-  int rep_fncol; // number of report day effects to apply
-  matrix[rep_fnrow, rep_fncol + 1] rep_fdesign; // design matrix for report dates
-  int rep_rncol; // number of standard deviations to use for pooling for rds
-  matrix[rep_fncol, rep_rncol + 1] rep_rdesign; // Pooling pmf design matrix 
+  int rep_fnrow; 
+  array[g, rep_t] int rep_findex; 
+  int rep_fncol;
+  matrix[rep_fnrow, rep_fncol + 1] rep_fdesign;
+  int rep_rncol; 
+  matrix[rep_fncol, rep_rncol + 1] rep_rdesign; 
+  array[2] real rep_beta_sd_p;
+
+  // Missing reference date model
+  int model_miss;
+  int miss_fnindex;
+  int miss_fncol;
+  int miss_rncol;
+  matrix[miss_fnindex, miss_fncol + 1] miss_fdesign;
```

Comment:
We don't have it here as we don't need it because `miss_` is not sparse and so no lookup between rows and `_fnindex` is needed. That definitely needs to be better/at least a bit documented. We could have it regardless but it just wouldn't be used.

Reply:
Okay, I think it is better to not have it here to avoid overhead, but it would be good if we could add a comment about this somewhere. Maybe just a "sparse specification" / "dense specification" comment at the beginning of each block?

Reply:
I've made an issue for this (#153) and suggest we deal with it in another PR with additional doc changes in the R code to make the general approach as clear as possible.